Update generator file system IO methods. Closes #107

peterblazejewicz opened the pull request:
Hello,

As reported on: https://github.com/OmniSharp/generator-aspnet/issues/107

This commit replace file system touching operations with
Yeoman's wrapper around mem-fs module as advised by Yeoman's documentation.
The commit removes use of already deprecated api, use this.fs module wrapper
and fixes problem with relative paths used in base class.
Tested with both npm test and current e2e test
The npm test have been failing - unless I rewrote implementation in base generate to use `this.templatePath` and `this.destinationPath` as per current docs.

Tests run locally are OK (`84 passing (739ms)` etc), but I've run them only on OS X machine.

http://yeoman.io/authoring/file-system.html

@sayedihashimi @spboyer Can you guys review this PR and test on other OS? (I'll try to verify on Windows OS at my work, but can't promise :) )

Thanks!

spboyer commented:
@peterblazejewicz the tests are failing, can you look into this and resubmit?

sayedihashimi commented:
When I ran tests locally I was receiving an error on OSX.

```
 Subgenerators without arguments tests
    aspnet:PackageJson
      1) "before all" hook

  0 passing (42ms)
  1 failing

  1) Subgenerators without arguments tests aspnet:PackageJson "before all" hook:

      Uncaught AssertionError: Trying to copy from a source that does not exist: package.json
      + expected - actual

      +true
      -false

      at Generator.generateStandardFile (script-base-basic.js:19:11)
      at Generator.createItem (PackageJson/index.js:13:7)
```

I upgraded yeoman to ```~0.19``` in package.json ran, ```npm install``` and then ```npm test``` to see if that made a difference but it didn't.

peterblazejewicz commented:
@sayedihashimi @spboyer 
updated, rebased, squashed.
The problem was not only with deprecation - but with updated Yo dependencies. I've made a lot of updates in implementation - but without removing or refactoring everything. 
I've just followed Yo/Yo-Generator release notes and:
- updated our `package.json` to match updated dependencies
- went through api step by step and updated methods calls to new syntax
- used new methods
- removed old methods.
Etc.
Please make sure to:
- update your loal Yo via `npm update -g yo`
- update your local, repository `node_modules`. Best to drop them and simply run again:
```
npm install
```

Thanks!

spboyer commented:
@peterblazejewicz @sayedihashimi LGTM, Sayed can you confirm this as well and we can merge?

sayedihashimi commented:
+1 thanks

sayedihashimi commented:
Go ahead and merge if its working. I hadn't had a chance to try and MSFT not for at least a few days.

spboyer commented:
Merged! Thanks.
